api/src/main/java/net/md_5/bungee/api/event/ServerConnectEvent.java

```diff
@@ -61,10 +61,20 @@ public ServerConnectEvent(ProxiedPlayer player, ServerInfo target, Reason reason
          * Execution of a command
          */
         COMMAND,
+        /**
+         * Redirecting to another server when client loses connection to server
+         * due to an exception.
+         */
+        SERVER_DOWN_REDIRECT,
         /**
          * Redirecting to another server when kicked from original server.
          */
         KICK_REDIRECT,
+        /**
+         * Redirecting to another server when kicked from server during the
+         * handshake.
+         */
+        KICK_REDIRECT_HANDSHAKE,
```

Comment:
Why does this need to be distinguished?

Reply:
Usually kicks during the login are bans, whitelists, etc. Just allows for more granular control really.

Reply:
Yeah but it also complicates the API usage (eg in switches) for granular control that I'm not sure is required

Reply:
Okay, I reverted the change.